packages/financial-templates-lib/src/price-feed/DominationFinancePriceFeed.js

```diff
@@ -0,0 +1,240 @@
+  }
+
+  _makeHistoricalPricesUrl(pair) {
+    return `${this.BASE_URL}/price/${pair}/history`;
+  }
+
+  getCurrentPrice() {
+    return this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
+    if (this.lastUpdateTime === undefined) {
+      return undefined;
+    }
+
+    // Set first price period in `historicalPrices` to first non-null price.
+    let firstEntry;
+    for (const row of this.historicalPricePeriods) {
+      if (row && row.openTime && row.closeTime && row.closePrice) {
+        firstEntry = row;
+        break;
+      }
+    }
+
+    // If there are no valid price periods, return null.
+    if (!firstEntry) {
+      return null;
+    }
+
+    // If the time is before the first piece of data in the set, return null because
+    // the price is before the lookback window.
+    if (time < firstEntry.openTime) {
+      return null;
+    }
+
+    // `historicalPrices` are ordered from oldest to newest.
+    // Find the first entry that is after the requested timestamp
+    const match = this.historicalPricePeriods.find(pricePeriod => {
+      return time < pricePeriod.closeTime;
+    });
+
+    // If there is no match, that means that the time was past the last data point.
+    // In this case, the best match for this price is the current price.
+    if (match === undefined) {
+      let returnPrice = this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+      if (verbose) {
+        const priceDisplay = this.convertDecimals(returnPrice.toString());
+        const priceUrl = this._makePriceUrl(this.pair);
+
+        console.group(`\n(${this.pair}) No historical price available @ ${time}`);
+        console.log(`- ✅ Time is later than earliest historical time, fetching current price: ${priceDisplay}`);
+        console.log(
+          `- ⚠️  If you want to manually verify the specific prices, you can make a GET request to: \n- ${priceUrl}`
+        );
+        console.groupEnd();
+      }
+      return this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+    }
+
+    let returnPrice = this.invertPrice ? this._invertPriceSafely(match.closePrice) : match.closePrice;
+    if (verbose) {
+      const url = this._makeHistoricalPricesUrl(this.pair);
+
+      console.group(`\n(${this.pair}) Historical Prices @ ${match.closeTime}`);
+      console.log(`- ✅ Price: ${this.convertDecimals(returnPrice.toString())}`);
+      console.log(
+        `- ⚠️  If you want to manually verify the specific exchange prices, you can make a GET request to: \n  - ${url}`
+      );
+      console.log(
+        '- This will return the historical prices as "data.rows". Each row contains: [unix_timestamp, price].'
+      );
+      console.log("- Note that you might need to invert the prices for certain identifiers.");
+      console.groupEnd();
+    }
+    return returnPrice;
+  }
+
+  getHistoricalPricePeriods() {
```

Comment:
Thanks! Just to be clear, at the moment the API only returns the minutely, 72 hour lookback. We internally have a separate system that handles arbitrary tick size and arbitrary lookback for charting purposes that isn't publicly documented (yet) since we've been still making adjustments here and there.

This would be a separate task/issue that we can tackle before UMA needs to run the numbers for the mining rewards, but if you know offhand what tick size and lookback would be necessary for the dev mining rewards, we likely can adjust our public API accordingly to support it and make a separate PR. Worst case, we can run some ad-hoc SQL for you lol. If this requires more discussion, we can always loop back in the Discord. Just wanted to give full disclosure regarding that.

Reply:
ok, that's not a problem. right now if your tokens are listed on coingecko then it's also fine. You have them listed there, right?

In the future, it would be ideal to expand this method to be able to return data looking further back. Regarding the tick size and lookback this is somewhat unimportant for the devmining. As long as there is some arbitrarily high precision data feed that can query up to a week in the past it should be fine. The Devmining is designed to work with different tick size data feeds.

Reply:
@chrismaree I've updated the PR to pull from a newly deployed API of ours that supports adjustable lookback and tick size. The only thing to keep in mind is that it will return a maximum of 5,000 rows per request so you likely will need to adjust the `tickPeriod` parameter to say 5 min = 300 secs (~2,000 rows for 1 week) and `lookback` as appropriate. Figured best to just take care of this now 😅 ...

Let me know if the PR is good to go! 👍🏾

Reply:
Ok cool! will try it out. Should not block this PR though :)